guide/event-handling/README.md

```diff
@@ -0,0 +1,144 @@
+
+The callback function passed takes argument(s) returned by its respective event, collects them in an `args` array using the `...` [rest parameter syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters), then calls `event.execute` function while passing in the `args` array using the `...` [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). They are used here because different events in discord.js have different numbers of arguments. The rest parameter collects these variable number of arguments into a single array, and the spread syntax then takes these elements and passes them to the `execute` function.
+
+After this, listening for other events is as easy as creating a new file in the `events` folder. The event handler will automatically retrieve and register it whenever you restart your bot.
+
+## Passing `Client` to event files
+
+You may have noticed how important the `Client` class is. You created a `client` instance of this class in the `index.js` file. Most of the time you can use this `client` instance in other files by either obtaining it from one of the other discord.js structures or from function parameters. In your `message` event, you can use `message.client`. When you don't have access to any of the structures that have the `client` property, you'll have to use the latter method. The prime example of this is the `ready` event.
+
+The `ready` event does not have arguments, meaning that `args` will be an empty array, thus nothing will be passed to the `execute` function in `ready.js`. To obtain the `client` instance, you'll have to pass it as an argument along with the `args` array in the event handler. Back in `index.js`, make the following changes:
+
+```diff
+for (const file of eventFiles) {
+	const event = require(`./events/${file}`);
+	if (event.once) {
+-		client.once(event.name, (...args) => event.execute(...args));
++		client.once(event.name, (...args) => event.execute(...args, client));
+	} else {
+-		client.on(event.name, (...args) => event.execute(...args));
++		client.on(event.name, (...args) => event.execute(...args, client));
+	}
+}
+```
+
+This allows `client` to be available as the **last** argument to the `execute` function in each event file. You can make use of `client` in `ready.js` by logging your bot's tag in the console when it becomes ready:
+
+```diff
+module.exports = {
+	name: 'ready',
+    once: true,
++	execute(client) {
+-		console.log('Ready!');
++		console.log(`Ready! Logged in as ${client.user.tag}`);
+	},
+};
+```
+
+You can omit the `client` argument from the `execute` function in files where you don't need it. For example, it isn't needed in the `message.js` file because its first argument is a `Message` instance, meaning you can use `message.client`.
```

Comment:
```suggestion
::: tip
You can omit the `client` argument from the `execute` function in files where you don't need it. For example, it isn't needed in the `message.js` file because its first argument is a `Message` instance, meaning you can use `message.client`.
:::
```